### ISSUE TITLE
Performance in my game is awful on base graphics configuration

### ISSUE THREAD

**Issue by CharleneMaure** [TheGameCreators/GameGuruRepo#4431]
https://drive.google.com/file/d/1FH4OMe7aIycHTI-lw2eWmzoduFCca7LL/view?usp=share_link

**Comment by MonkeyFrogStudio** [comment]
What are your system specs? What screen resolution are you running the game at? It's really, really difficult for the devs to help you or see what's going on with this totally relative information.

**Comment by davetgc** [comment]
I'm sorry you're having issues, but can you please let use know what engine is this for? Is it MAX or Classic?

As mentioned, can you also please tell us your system specs.

Thanks

**Comment by CharleneMaure** [comment]
Gtx 3050 Ti

I9 12-12900H

16GB Ram

**Comment by davetgc** [comment]
And this is for Classic or MAX?

Thanks,

**Comment by CharleneMaure** [comment]
Max

**Comment by LukeTheLukeLTL** [comment]
Yes I confirm that there are performance issues.
A few days ago I released the demo of my video game, and some users reported me issues of stuttering and in general of the low framerate, about 30 FPS. I personally did not have any major performance issues (My game runs between 50 and 70 FPS, which is still low in my opinion), but I did notice quite a lot of stuttering especially during fights.
I have an RTX 3070 though, so I have an above average PC.
GGMAX currently can't handle anything beyond a simple tech demo, unless you have a high-end PC.
I think it's because of the lack of, for example, LODs and texture streaming. Then the games made with GGMAX use too much RAM, even for a simple level 16GB is used, and if you want to make a serious level with lots of objects you go over 20GB instead, and that is really too much and is a sign of poor optimization of the engine.
We have to remember that the average gamer still has a GTX 1060, or a 1650, so now that GGMAX is out of Early Access it's essential that it be optimized for such PCs as well.

**Comment by LeeBamberTGC** [comment]
Just to put a first stake in the ground for optimization, let's say that all the HUB DEMOS in MAX should run at 60 fps on the advertised minimum specification:

MINIMUM:
Requires a 64-bit processor and operating system
OS: Windows 10 or newer (64-bit versions only)
Processor: Intel Core i5-7400 | AMD Opteron 4274 HE, or better.
Memory: 16 GB RAM
Graphics: NVIDIA GeForce GTX 970 | AMD Radeon R9 Fury, with 4 GB VRAM, or better**

Naturally, if you drop in ten thousand characters in a big pile and and every single object from the library around that, you will not be expected to run at 60 fps on the minimum spec :)  Game developers VERY SOON realise that the performance of their games is not a given and you must craft your content for both storytelling, visuals, and performance.  Believe it or not, even Unity and Unreal games drop below 60 fps somehow!

**Comment by MonkeyFrogStudio** [comment]
@LeeBamberTGC Yes, however, here are my system specs:

Intel i9-10900k (10 cores, 20 threads) 3.7GHz (overclocked to reach up to 5Ghz)
64GB system RAM
Windows 11 (all up to date and 64-bit)
Nvidia 3090 ti with 25GB VRAM

While on my own interior level and the Cyberpunk Streets demo level, I get 100FPS (due to vsync - up to 300+ with it turned off, except on Cyberpunk Streets, which stays at about 120+), I get far, far less on outdoor levels. With any of the demos, like Island Showdown or Snowy Mountain Stroll, I get 60 - 80 (but mostly in the 60s and low 70s). So, clearly the engine isn't behaving like it's optimized (or well optimized). I can max out settings in other games, including newly released games, and get in excess of 100fps. Heck, I can run some games with raytracing turned on and get better frame rates than the outdoor demos in MAX.

One issue, which had been reported previously, is that things are being rendered that are behind walls, etc. For example, in the Offworld Station demo level, if you go to a room at one end of the level and face a wall that has other rooms on the other side (not seen, but blocked by a solid wall), then frame rates take a pretty big hit and draw calls climb. Yet, nothing should be seen behind these walls. So, optimization does not seem to be taking place as it should.

**Comment by LeeBamberTGC** [comment]
Thanks for the replies, all valid points and I agree, many optimizations will be needed before your game can compete with one created by an AAA team. These teams (including Unity and Unreal dev teams) include a specialist class of coder that focuses on engine-level performance and spend years perfecting their art and improving with each engine they work on. Alas, I am more of a general coder and do not have those skills.  I could do it, but it would take me a year if I did nothing else. If MAX does well commercially, I would hire a specialist who only focuses on optimization, but with an EA Steam review score of 71%, I do not think that is going to happen. Chicken and Egg, cannot make sales with a slow engine and need a fast engine to make sales :)

The current plan is for optimization to happen as we plod along. Are there suggestions on how we could solve this in a shorter time frame than Lee's usual rate of progress?

**Comment by SampeyFS** [comment]
low Fps ---> invidia RTX 3060 Ti  16 GB Ram  Windows 11

**Comment by Kravenwolf** [comment]
@LeeBamberTGC, I'm running a GTX 1060 (3GB), which is within ~2% performance of the minimum spec 970 according to UserBench (see "average score"):
https://gpu.userbenchmark.com/Compare/Nvidia-GTX-970-vs-Nvidia-GTX-1060-3GB/2577vs3646

It's also a variant of the most popular card on STEAM. I also have an "average" 4core/4thread, non-overlocked 7600K intel CPU and 16GB of 2133 RAM.

So, I'm avail. for some benchmark testing of the demo maps if need be.

Just fired up the snowy mountain stroll demo on the march build and averaged ~29FPS during shootouts. Frames jumped up into the mid-40s or low 50s during the quieter, less bloody parts.

![snowy](https://user-images.githubusercontent.com/103399301/231013815-9e0bb2ff-fbef-48bb-a268-4caae5cc3e4d.jpg)


cannot make sales with a slow engine and need a fast engine to make sales :)

**Comment by LeeBamberTGC** [comment]
@Kravenwolf Thanks for the stats, step one is to get these figures to 60 fps no matter what is happening in the game :)

**Comment by 3coms** [comment]
Ignore if it has to do, but it looks like GGMax does not refresh the mem between levels.
I ran my level and of course, I've got 0,2 fps and 21 physic ram waste, obviously impossible to do anything like that; so, I closed my level and open a new one with just a scroll note mesh, then I'0ve got 14 GBD physic ram waste, so to complete the experiment, I close GGMax>Open afain>run the scroll note level, and I've got just 3 gbs physic ram waste, with the same level; it's telling me that GGMax is not refreshing.
![GGMax_EXP_no_Refreshing_MEM_pic02](https://user-images.githubusercontent.com/47690174/232170880-615ba0c9-fc57-44c5-b209-891a76365732.png)
![GGMax_EXP_no_Refreshing_MEM_pic91](https://user-images.githubusercontent.com/47690174/232170881-4be38fc3-ed42-44ac-98aa-34da9c8776c0.png)

**Comment by supe313** [comment]
@LeeBamberTGC as a hellish coder and creator of ranger guy  (3d game editor) its another easy way to do the job. recreate ggMax in unity. its totally possible and i even release v14 of my tool and it works very well and cool. all things you must know it is you are wasting your time and money in this direction , change your direction!
if you have any question to how do this send a message to my old gmail account: hamedsheygh@gmail.com
Best wishes for you and your team

**Comment by SampeyFS** [comment]
It is an anomalous thing that an RTX 3060 Ti 8 Gb does so few FPS I understand that your level grows but drops too much it must be fixed because it brings the GPU to high temperatures

**Comment by MonkeyFrogStudio** [comment]
@SampeyFS I have a 3090ti and get FPS in the hundreds on an empty level and fps over 300+ on my own levels on average. The exterior levels, like Snowy Mountain Stroll, get 60-80fps. So, if you, with a 3060ti, are getting really low FPS and your GPU is hot, then there seems to be another issue at play here. MAX definitely needs to be optimized, but the issues you're experiencing seem a bit unique, even for MAX.
